Python/context.c

```diff
@@ -108,6 +108,79 @@ PyContext_CopyCurrent(void)
     return (PyObject *)context_new_from_vars(ctx->ctx_vars);
 }
 
+static const char *
+context_event_name(PyContextEvent event) {
+    switch (event) {
+        #define CASE(op)                   \
+        case PY_CONTEXT_EVENT_##op:         \
+            return "PY_CONTEXT_EVENT_" #op;
+        PY_FOREACH_CONTEXT_EVENT(CASE)
+        #undef CASE
+    }
+    Py_UNREACHABLE();
+}
+
+static void notify_context_watchers(PyContextEvent event, PyContext *ctx)
+{
+    assert(Py_REFCNT(ctx) > 0);
+    PyInterpreterState *interp = _PyInterpreterState_GET();
```

Comment:
While this is incredibly minor, please add an early exit to this function. I'd like the "notify" mechanism to have as low overhead as theoretically possible.

Reply:
What did you have in mind?   if !bits return? 
I'm not seeing it.

Also I'm pretty much copying in totality the following existing code from PyCodeObjects https://github.com/python/cpython/blob/main/Objects/codeobject.c#L32